### PR TITLE
Pull Request: JavaScript Assignment - 1 Solution

### PR DESCRIPTION
**Pull request by avanish-syvora**
Summary

This PR adds standalone JavaScript solutions for a set of core programming problems involving:

1. Finding the second largest number in an array
2. Calculating frequency of lowercase alphabets in a string
3. Flattening a nested JavaScript object
4. Unflattening a flat key-value object into nested structure

 Files Added

| File Name               | Description                                                      |
| ----------------------- | ---------------------------------------------------------------- |
| `secondlargest.js`      | Returns second largest number without using `.sort()`            |
| `frequencycalc.js` | Computes frequency of each lowercase alphabet using arrays       |
| `flatten.js`            | Converts deeply nested JS object into a flattened key-value pair |
| `Unflatten.js`          | Reconstructs nested object from a flat dot-notation object       |

---
Highlights

* Modular, reusable, and well-structured code
* Edge cases handled (e.g. empty input, duplicates, invalid types)
* Clean variable naming and function boundaries
* Adheres to functional principles and consistent syntax
* Minimal dependencies – only vanilla JavaScript

---
Testing

Each file includes:

* Inline or block-based test cases using `console.log()`
* Covers edge conditions like empty inputs, nested structures, duplicates, etc.

---

Please let me know if there are any req changes!

### CODE REVIEW

File: flatten.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+function flatten(obj) {
+  let result = {};
+
+  function helper(curr, path) {
```
Comment (chainguy2049) [inline flatten.js]: Add comments to describe how this function works.

File: flatten.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    for (let key in curr) {
+      let newPath = path ? path + '.' + key : key;
+      if (typeof curr[key] === 'object' && curr[key] !== null && !Array.isArray(curr[key])) {
+        helper(curr[key], newPath);
+      } else {
+        result[newPath] = curr[key];
```
Comment (chainguy2049) [inline flatten.js]: Check indentation and style guide for js
